x-pack/elastic-agent/pkg/composable/providers/host/host.go

```diff
@@ -0,0 +1,99 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package host
+
+import (
+	"os"
+	"reflect"
+	"runtime"
+	"time"
+
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/core/logger"
+	"github.com/elastic/go-sysinfo"
+
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/composable"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/config"
+)
+
+// DefaultCheckTimeout is the default timeout used to check if any host information has changed.
+const DefaultCheckTimeout = 5 * time.Minute
+
+func init() {
+	composable.Providers.AddContextProvider("host", ContextProviderBuilder)
+}
+
+type infoFetcher func() (map[string]interface{}, error)
+
+type contextProvider struct {
+	logger *logger.Logger
+
+	// used by testing
+	checkTimeout time.Duration
+	fetcher      infoFetcher
+}
+
+// Run runs the environment context provider.
+func (c *contextProvider) Run(comm composable.ContextProviderComm) error {
```

Comment:
I wonder if we should rather call this `Start` instead of `Run` as it is not blocking. We kind of had a convention on this in Beats as we kept mixing the two. @urso might remember?

Reply:
I called it `Run` because its up to the provider, and `Run` comes from the interface that the host provider must implement. In this case this provider is asynchronous so it uses the `<-comm.Done()` to know when it needs to stop its created go routine.